src/main/java/fruitshop/strategy/operation/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,18 @@
+package fruitshop.strategy.operation.impl;
+
+import fruitshop.model.Operation;
+import fruitshop.strategy.operation.OperationHandler;
+import fruitshop.strategy.operation.OperationStrategy;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<Operation, ? super OperationHandler> handlers;
```

Comment:
do you really need generics here?

Reply:
Yes. It's usage of kind of polymorphism. "Consumer"